sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -623,11 +624,62 @@ public void testInsertFromQueryOnDuplicateKey() throws Exception {
         assertThat((Long) response.rows()[0][1], is(2L));
     }
 
-
+    @Test
     public void testInsertFromSubQueryWithVersion() throws Exception {
         expectedException.expect(SQLActionException.class);
         expectedException.expectMessage("\"_version\" column is not valid in the WHERE clause");
         execute("create table users (name string)");
         execute("insert into users (name) (select name from users where _version = 1)");
     }
+
+    @Test
+    public void testInsertFromSubQueryPartitionedTableCustomSchema() throws Exception {
+        execute("create table custom.source (" +
+                "  name string, " +
+                "  zipcode string, " +
+                "  city string" +
+                ") partitioned by (city) with (number_of_replicas=0)");
+        execute("create table custom.destination (" +
+                "  name string, " +
+                "  zipcode string, " +
+                "  city string" +
+                ") partitioned by (zipcode) with (number_of_replicas=0)");
+        ensureGreen();
+        execute("insert into custom.source (name, zipcode, city) values (?, ?, ?)", new Object[][]{
+                {"Schulz", "10243", "Berlin"},
+                {"Dings", "14713", "Leipzig"},
+                {"Foo", "10243", "Musterhausen"}
+        });
+        ensureGreen();
+        refresh();
+
+        execute("select * from custom.source order by zipcode");
```

Comment:
jep, checking (quite verbose) that what is inside that table gets inserted (both by values and by subquery) correctly